Moving Photino.Blazor to Hosting

Denny09310 opened the pull request:
I managed to implement the IHostBuilder and IHostApplicationBuilder interfaces into the PhotinoBlazorAppBuilder. It's a bit of a hack job for now, though, mostly copy/pasting with a sprinkle of tweaks from the .NET source, just to get something going.

This can be useful for all the Hosting extensions and/or internal services that Microsoft.Extensions.Hosting offers and a way to register them gracefully inside the photino builder

kimdiego2098 commented:
is this work feasible?

Denny09310 commented:
> is this work feasible?

Yes. I've used it successfully in one of my projecrs without encountering errors (for now 😁)

MikeYeager commented:
@Denny09310 We see what you're doing here and it looks like a nice addition. We're curious why you copied .NET source code into here instead of referencing the .NET code? _"I managed to implement the IHostBuilder and IHostApplicationBuilder interfaces into the PhotinoBlazorAppBuilder"_

Denny09310 commented:
Unfortunately, most of the functionality is internal to `Microsoft.Extensions.Hosting`. I may have overlooked something, but so far, I haven’t found a way to integrate those internal classes into the `Photino.Blazor` project while maintaining full access to their features. The main drawback of copying these files is the need to keep them in sync with the originals. Ideally, it would be better to find a way to access those classes without duplicating them.

Denny09310 commented:
@MikeYeager, after a thorough investigation, I've discovered that the solution was simpler than expected. I removed all the .NET source code files and implemented both the `IHostApplicationBuilder` and `IHost` within the Photino* classes. I ran all the sample applications, and everything appears to be working, though it likely requires more testing—especially concerning the `IFileProvider` request noted in previous issues. Currently, it uses the classic Blazor strategy (searching for the `wwwroot`), but I'm not sure if that behavior can be overridden.

MikeYeager commented:
@Denny09310 We've merged this into a Feature branch for testing, but we can't get it to run. 
<img width="947" height="889" alt="image" src="https://github.com/user-attachments/assets/10d81419-336b-4449-8b62-07ba223a52e0" />

We're wondering how this is open to all URLS. t appears to be restricted to localhost.
<img width="1039" height="1081" alt="image" src="https://github.com/user-attachments/assets/99cd71f8-d351-4381-a932-f8f0d861eaa7" />

MikeYeager commented:
@Denny09310 Any ideas on resolving this issue so we can press on with a merge? Thx.

Denny09310 commented:
@MikeYeager I honestly don't know why. I suppose you are using the sample project 'HelloWorld'. I think the builder from ASP .NET expects at least to have a Router for navigation and an index page. I'm not entirely sure but using the other samples works as expected, only that one gives this problem. Adding the router to `App.razor` fixed it.

For the other request, I don't know what are you meaning as I've only cleaned up the PhotinoWebViewManager file due to code-maid being active without changing any logic, if I remember correctly
